samples/client.c

```diff
@@ -492,12 +491,23 @@ access_region(int sock, int region, bool is_write, uint64_t offset,
         err(EXIT_FAILURE, "failed to alloc recv_data");
     }
 
-    pthread_mutex_lock(&mutex);
-    ret = tran_sock_msg_iovec(sock, msg_id--, op,
-                              send_iovecs, nr_send_iovecs,
-                              NULL, 0, NULL,
-                              recv_data, recv_data_len, NULL, 0);
-    pthread_mutex_unlock(&mutex);
+    /*
+     * For all accesses that aren't writes to BAR1, use the standard method.
+     * For writes to BAR1 (which come from the fake guest thread), we send
+     * the command with the `no_reply` flag set so we don't interfere with
+     * the main thread. This means we don't require a mutex.
+     */
+    if (region != 1 || !is_write) {
+        ret = tran_sock_msg_iovec(sock, msg_id--, op,
```

Comment:
I think it would be a bit weird to add the `no_reply` argument to `tran_sock_msg_iovec` given that all the `tran_sock_msg_*` functions send a message, wait for a reply, and put the reply in a given buffer? We use `tran_sock_send_iovec` for the `no_reply` one because it only does the sending part and doesn't wait for a reply.

Reply:
Agreed, I prefer what is here already